### PR TITLE
New function for interim fields formatting

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses

This Pull Request adds a function to make the formatting of interim fields easier. It also adds a new function `to_list` in the api

## Current behavior before PR

Formatting values and units of interim fields for human visualization is cumbersome

## Desired behavior after PR is merged

Formatting values and units of interim fields for human visualization made easy

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/bika/lims/api/__init__.py [hunk 1]
```diff
@@ -1787,3 +1788,21 @@ def is_string(thing):
...
+    if is_string(value):
+        try:
+            val = json.loads(value)
+            if isinstance(val, (list, tuple, set)):
+                value = val
+        except (ValueError, TypeError):
+            pass
+    if not isinstance(value, (list, tuple, set)):
+        value = [value]
+    return list(value)
```
Comment (ramonski) [inline src/bika/lims/api/__init__.py:1818]: Please split this function into 2, e.g. something like this:
```python
def parse_json(thing, default=""):
    """Parse from JSON
    """
    try:
        return json.loads(thing)
    except (TypeError, ValueError):
        return default

def to_list(value):
    """Converts the value to a list
    :param value: the value to be represented as a list
    :returns: a list that represents or contains the value
    """
    if is_string(value):
        value = parse_json(value, [])
    if not isinstance(value, (list, tuple, set)):
        value = [value]
    return list(value)
```

**Review by ramonski — CHANGES_REQUESTED** [review]

**Comment by ramonski** [comment]
Only a minor thing to increase reusability

File: src/bika/lims/api/__init__.py (hunk 1 above)
Comment (xispa) [reply src/bika/lims/api/__init__.py:1818]: Done with https://github.com/senaite/senaite.core/pull/2365/commits/4134d9b0a0dd2fdabe58307657460a077fef5a85

**Review by ramonski — APPROVED** [review]

**Comment by grulisco** [comment]
:heart:

**Pull request merged**